packages/browser/src/login/oidc/redirectHandler/AuthCodeRedirectHandler.ts

```diff
@@ -86,23 +85,26 @@ export default class AuthCodeRedirectHandler implements IRedirectHandler {
   ) {}
 
   async canHandle(redirectUrl: string): Promise<boolean> {
-    const url = new URL(redirectUrl, true);
-    return (
-      url.query !== undefined &&
-      url.query.code !== undefined &&
-      url.query.state !== undefined
-    );
+    try {
+      const myUrl = new URL(redirectUrl);
+      return (
+        myUrl.searchParams.get("code") !== null &&
+        myUrl.searchParams.get("state") !== null
+      );
+    } catch {
+      throw new Error(
+        `[${redirectUrl}] is not a valid URL, and cannot be used as a redirect URL.`
```

Comment:
Can't we catch the actual error thrown (you just have `} catch {` there now), and report the description within that exception in the message here (I don't think it would be a security problem to report that)...?

Reply:
I think the additional context is more useful to a developer. (That said, eventually we'll want to consider typed errors instead of strings, and preferably use strings from a vocab.)

Reply:
Typed errors sound like a great idea, we should definitively right a couple of stories to have that in the backlog. The Java LIT has an Exception support module if I recall correctly, that's something that we might want to look into, because here the additional context is rather poor (`TypeError: Invalid URL: beep boop I am a robot`), which is why I wasn't sure about adding it in the first place ^^

Reply:
Resolved in 413e960a5c9fe7aca0ffad42c68d62293196dc25